### PR TITLE
Feature/fetch listings

### PR DESCRIPTION
**Pull request by collinkallery**
### Type of change made:
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Styling (Styling to new or existing files)

### Detailed Description
- Render and route paths for listing container 
- Fetch of an area's listings once listing container is rendered
- Initial display of individual listings' names
- Using an Object.keys to iterate over the "details" object for each listing, but having some trouble getting it working just right. Currently, it's showing the information for every listing, under each listing. Meaning for listing 1, info for listing 1, 2, 3, 4, etc are shown under listing 1, and the same for listings 2, 3, and 4. 
- Very basic styling for listing container and listing 

### Why is this change required? What problem does it solve?
- This allows a user to navigate to the areas page, click on a given area and be routed to the listings for that given area. 

### Was any previously written code changed?
- Only thing changed was adding an "exact" path to the areas container so that when an area was chosen, it showed the listing container only, and not the listing container AND the areas container. 

### Directory modified:
- [x] - App
- [ ] - Login
- [x] - Area
- [x] - AreaContainer
- [x] - Listing
- [x] - ListingContainer

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -10,19 +11,22 @@ import {Route, Redirect} from "react-router-dom";
 class App extends Component {
   constructor() {
     super();
+    this.mounted = false;
```
Comment (collinkallery) [inline src/components/App/App.js:14]: Read an article about this - used for "cleanup" to cancel fetch requests when a component is unmounted. Helps with memory leaks.

File: src/components/App/App.js [hunk 2]
```diff
@@ -67,8 +74,18 @@ class App extends Component {
...
+          path='/Areas/:id/Listings'
+          exact
+          render={({match}) => {
+            const { id } = match.params;
+            const areaToRender = this.state.areas.find(area => {
+              return area.id === parseInt(id)
+            });
+            return <ListingContainer {...areaToRender} />
+        }} />
+        <Route exact path="/Areas" render={() => <AreaContainer areas={this.state.areas} />} />
```
Comment (collinkallery) [inline src/components/App/App.js:87]: Exact path for areas to show only the areas container when at this link.

File: src/components/Area/Area.js [hunk 3]
```diff
@@ -1,14 +1,56 @@
...
 import './Area.css';
+import { Link } from "react-router-dom";
 
 const Area = (props) => {
   return (
     <section className="area-card">
       <h3 className="area-header">{props.name} ({props.area})</h3>
       <p className="area-about">{props.about}</p>
-      <button id={props.id} className="see-listings-button">View Listings</button>
+      <Link to={`/Areas/${props.id}/Listings`}>
```
Comment (collinkallery) [inline src/components/Area/Area.js:10]: Navigates to a given area's listings.

File: src/components/AreaContainer/AreaContainer.js [hunk 4]
```diff
@@ -17,13 +19,32 @@ const AreaContainer = ({areas}) => {
         region_code={area.region_code}
         quick_search={area.quick_search}
         listings={area.listings}
+        displayListings={displayListings}
+        currentListings={currentListings}
```
Comment (collinkallery) [inline src/components/AreaContainer/AreaContainer.js:23]: This is not being used (I believe) and can likely be removed.

File: src/components/AreaContainer/AreaContainer.js [hunk 5]
```diff
@@ -17,13 +19,32 @@ const AreaContainer = ({areas}) => {
...
       />
     )
   })
 
   return (
     <section className="areas-container">
       {areasToDisplay}
+      {/*<Route
```
Comment (collinkallery) [inline src/components/AreaContainer/AreaContainer.js:31]: Had moved the route to area container, but ran into some issues and remembered that Robbie said the routes should live in App, so moved it to App and all current routes are working.

File: src/components/Listing/Listing.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import './Listing.css';
+
+const Listing = (props) => {
+
+  const renderListingDetails = () => {
```
Comment (collinkallery) [inline src/components/Listing/Listing.js:6]: Breaking change: tried to use Object.keys to iterate over specifically a given listing's "details" object (where beds, baths, cost per night are). But having some weird behavior, its showing the information for EVERY listing, underneath every single listing (if that makes sense - more detailed descrip / example in the PR).

File: src/components/ListingContainer/ListingContainer.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  listingsToDisplay = () => {
+    let listings = this.state.listingData.map(listing => {
+      return (
+        <Listing {...listing} />
+      )
+    })
+    return listings;
+  }
+
+  componentDidMount() {
```
Comment (collinkallery) [inline src/components/ListingContainer/ListingContainer.js:23]: Fetch for listings.

**Comment by C-Bandstra** [comment]
Im going to try and work out the object.keys issue.

**Pull request merged**